build-package.sh

```diff
@@ -1,6 +1,21 @@
 #!/bin/bash
 # shellcheck disable=SC1117
 
+# Fixing the number of calls and the list of compiled packages
+if [[ ! "$TERMUX_BUILD_PACKAGE_CALL_DEPTH" =~ ^[0-9]+$ ]]; then
+	export TERMUX_BUILD_PACKAGE_CALL_DEPTH=0
+	export TERMUX_BUILD_PACKAGE_CALL_BUILT_PACKAGES_LIST=" "
+else
+	export TERMUX_BUILD_PACKAGE_CALL_DEPTH=$((TERMUX_BUILD_PACKAGE_CALL_DEPTH+1))
+fi
+
+# Adds a package to the list of built packages if it is not in the list
+termux_add_package_to_built_packages_list() {
+	if [[ "$TERMUX_BUILD_PACKAGE_CALL_BUILT_PACKAGES_LIST" != *" $1 "* ]]; then
```

Comment:
read from file to get current value and echo new value to file here instead of elswheree

Reply:
You haven't even updated file when dependencies get built with current single `echo` in `build-sh`.